internal/services/securitycenter/security_center_setting_resource.go

```diff
@@ -104,22 +103,26 @@ func resourceSecurityCenterSettingRead(d *pluginsdk.ResourceData, meta interface
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.SettingID(d.Id())
+	id, err := settings.ParseSettingID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	// TODO: switch to back when Swagger/API bug has been fixed:
-	// https://github.com/Azure/azure-sdk-for-go/issues/12724 (`Enabled` field missing)
-	resp, err := azuresdkhacks.GetSecurityCenterSetting(ctx, client, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
 		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
-	if properties := resp.DataExportSettingProperties; properties != nil {
-		d.Set("enabled", properties.Enabled)
+	if resp.Model != nil {
+		if alertSyncSettings, ok := (*resp.Model).(settings.AlertSyncSettings); ok && alertSyncSettings.Properties != nil {
+			d.Set("enabled", alertSyncSettings.Properties.Enabled)
+		}
+		if dataExportSettings, ok := (*resp.Model).(settings.DataExportSettings); ok && dataExportSettings.Properties != nil {
+			d.Set("enabled", dataExportSettings.Properties.Enabled)
```

Comment:
I was thinking that only one of them could ever be successfully cast to a matching model so then it would only get set one way. If that's a misunderstanding and both could be cast then perhaps I could check both simultaneously and error if they don't match